astropy/coordinates/matching.py

```diff
@@ -300,10 +300,11 @@ def search_around_sky(coords1, coords2, seplimit, storekdtree='_kdtree_sky'):
         The on-sky separation between the coordinates. Shape matches ``idx1``
         and ``idx2``.
     dist3d : `~astropy.units.Quantity`
-        The 3D distance between the coordinates. Shape matches ``idx1`` and
-        ``idx2``. If either ``coords1`` or ``coords2`` don't have a distance,
-        this is the 3D distance on the unit sphere, rather than a physical
-        distance.
+        The 3D distance between the coordinates. Shape matches ``idx1``
+        and ``idx2``; the unit is that of ``coords1``.
+        If either ``coords1`` or ``coords2`` don't have a distance,
+        this is the 3D distance on the unit sphere, rather than a
+        physical distance.
```

Comment:
Certainly good to ask, and the reason why I put in a PR, because there will be a bunch of corner cases like this that may warrant some discussion.
If we'd like to stick to the convention stated in the docstring here, there should indeed be a check for the distance of ``coords2`` when either the input (line 331) or the result (line 373) is empty.